addons/point_of_sale/models/pos_config.py

```diff
@@ -402,6 +404,9 @@ def _reset_default_on_vals(self, vals):
 
     def write(self, vals):
         self._reset_default_on_vals(vals)
+        if ('is_order_printer' in vals and not vals['is_order_printer']):
+            vals['printer_ids'] = [(5, 0, 0)]
```

Comment:
Can you use instead: https://github.com/odoo/odoo/blob/6ddb77094bb0654d6326c6f9dadbf4049b86f52d/odoo/fields.py#L4008